donkeycar/contrib/gradcam/cli.py

```diff
@@ -0,0 +1,432 @@
+    )
+
+    parser.add_argument(
+        '--video_name',
+        default='grad_cam_visualization',
+        help='Name for the output video file'
+    )
+
+    parser.add_argument(
+        '--fps',
+        type=int,
+        default=20,
+        help='Frames per second for the output video'
+    )
+
+    parser.add_argument(
+        '--stride',
+        type=int,
+        default=1,
+        help='Process every Nth frame'
+    )
+
+    parser.add_argument(
+        '--max_frames',
+        type=int,
+        default=None,
+        help='Maximum number of frames to process'
+    )
+
+    parser.add_argument(
+        '--alpha',
+        type=float,
+        default=0.4,
+        help='Transparency of the heatmap overlay (0.0-1.0)'
+    )
+
+    parser.add_argument(
+        '--exclude_regions',
+        type=str,
+        default=None,
+        help='Regions to exclude from heatmap (x1,y1,x2,y2|x1,y1,x2,y2|...)'
+    )
+
+    return parser.parse_args()
+
+
+def load_model_safely(model_path):
+    """
+    Load a Keras model with appropriate error handling.
+
+    Args:
+        model_path: Path to the model file
+
+    Returns:
+        Loaded Keras model
+    """
+    print(f"Loading model from {model_path}")
+
+    try:
+        # Try to load the model with custom objects to handle loss functions
+        custom_objects = {
+            'mse': tf.keras.losses.mean_squared_error,
+            'mean_squared_error': tf.keras.losses.mean_squared_error
+        }
+        model = load_model(model_path, custom_objects=custom_objects)
+        print("Model loaded successfully with custom objects")
+        return model
+    except Exception as e:
+        print(f"Error loading model with custom objects: {e}")
+
+        try:
+            # Alternative loading method with compile=False
+            model = load_model(model_path, compile=False)
+            print("Model loaded successfully with compile=False")
+            return model
+        except Exception as e2:
+            print(f"Fatal error loading model: {e2}")
+            sys.exit(1)
+
+
+def find_images(data_path):
+    """
+    Find image files in the data directory.
+
+    Args:
+        data_path: Path to the data directory
+
+    Returns:
+        List of image file paths
+    """
+    print(f"Searching for images in {data_path}")
+
+    # Different image patterns to try
+    patterns = [
+        os.path.join(data_path, "images/*.jpg"),
+        os.path.join(data_path, "**/images/*.jpg"),
+        os.path.join(data_path, "*.jpg"),
+        os.path.join(data_path, "**/*.jpg")
+    ]
+
+    for pattern in patterns:
+        print(f"Trying pattern {pattern}")
+        image_paths = sorted(glob.glob(pattern, recursive=True))
+        if image_paths:
+            print(f"Found {len(image_paths)} images with pattern {pattern}")
+            return image_paths
+
+    print("No images found with any pattern")
+    return []
+
+
+def load_metadata(data_path):
+    """
```

Comment:
Access to tub metadata and tub records is provided through the `Tub` class and hence this method will become obsolete.